include/nbl/video/IGPUCommandBuffer.h

```diff
@@ -227,6 +184,89 @@ class NBL_API IGPUCommandBuffer :
+    virtual bool clearAttachments(uint32_t attachmentCount, const asset::SClearAttachment* pAttachments, uint32_t rectCount, const asset::SClearRect* pRects) = 0;
+    virtual bool fillBuffer_impl(buffer_t* dstBuffer, size_t dstOffset, size_t size, uint32_t data) = 0;
+    virtual void bindVertexBuffers_impl(uint32_t firstBinding, uint32_t bindingCount, const buffer_t* const* const pBuffers, const size_t* pOffsets) = 0;
+    virtual bool dispatchIndirect_impl(const buffer_t* buffer, size_t offset) = 0;
+    virtual bool setEvent_impl(event_t* _event, const SDependencyInfo& depInfo) = 0;
+    virtual bool resetEvent_impl(event_t* _event, asset::E_PIPELINE_STAGE_FLAGS stageMask) = 0;
+    virtual bool waitEvents_impl(uint32_t eventCount, event_t* const* const pEvents, const SDependencyInfo* depInfos) = 0;
+    virtual bool copyBuffer_impl(const buffer_t* srcBuffer, buffer_t* dstBuffer, uint32_t regionCount, const asset::SBufferCopy* pRegions) = 0;
+    virtual bool copyImage_impl(const image_t* srcImage, asset::IImage::E_LAYOUT srcImageLayout, image_t* dstImage, asset::IImage::E_LAYOUT dstImageLayout, uint32_t regionCount, const asset::IImage::SImageCopy* pRegions) = 0;
+    virtual bool copyBufferToImage_impl(const buffer_t* srcBuffer, image_t* dstImage, asset::IImage::E_LAYOUT dstImageLayout, uint32_t regionCount, const asset::IImage::SBufferCopy* pRegions) = 0;
+    virtual bool blitImage_impl(const image_t* srcImage, asset::IImage::E_LAYOUT srcImageLayout, image_t* dstImage, asset::IImage::E_LAYOUT dstImageLayout, uint32_t regionCount, const asset::SImageBlit* pRegions, asset::ISampler::E_TEXTURE_FILTER filter) = 0;
+    virtual bool copyImageToBuffer_impl(const image_t* srcImage, asset::IImage::E_LAYOUT srcImageLayout, buffer_t* dstBuffer, uint32_t regionCount, const asset::IImage::SBufferCopy* pRegions) = 0;
+    virtual bool resolveImage_impl(const image_t* srcImage, asset::IImage::E_LAYOUT srcImageLayout, image_t* dstImage, asset::IImage::E_LAYOUT dstImageLayout, uint32_t regionCount, const asset::SImageResolve* pRegions) = 0;
+    virtual bool executeCommands_impl(uint32_t count, cmdbuf_t* const* const cmdbufs) = 0;
+    virtual bool regenerateMipmaps_impl(IGPUImage* img, uint32_t lastReadyMip, asset::IImage::E_ASPECT_FLAGS aspect) { assert(!"Invalid code path.");  return false; };
+
+private:
+    // Be wary of making it protected/calling it in the derived classes because it sets state which will overwrite the state set in base class methods.
+    inline bool checkForParentPoolReset()
+    {
+        if (m_cmdpool->getResetCounter() <= m_resetCheckedStamp)
+            return false;
+
+        m_resetCheckedStamp = m_cmdpool->getResetCounter();
+        m_state = ES_INITIAL;
+
+        m_segmentListHeadItr.m_cmd = nullptr;
+        m_segmentListHeadItr.m_segment = nullptr;
+        m_segmentListTail = nullptr;
+
```

Comment:
I think there should be a `checkForParentPoolReset_impl` here because how else will you nullify the segment list for backend-specific-tape?

Reply:
Yes, I think you're right, but should I do it, now that we'll be deprecating the GL backend?

Also, I can put a comment here to add a `checkForParentPoolReset_impl` when/if required for any new backend.

Reply:
> Yes, I think you're right, but should I do it, now that we'll be deprecating the GL backend?
> 
> Also, I can put a comment here to add a `checkForParentPoolReset_impl` when/if required for any new backend.

Again, think WebGPU.

However the reset counter is one for whole pool, not tape, and since you don't reset those separately... so in the end seems there's no need for a `checkForParentPoolReset`, only `releaseResourcesBackToPool_impl` which is already implemented.

Reply:
Resolved.